x-pack/plugins/ml/server/models/data_frame_analytics/models_provider.ts

```diff
@@ -89,30 +111,19 @@ export function modelsProvider(
      * Provides the ML nodes overview with allocated models.
      */
     async getNodesOverview(): Promise<NodesOverviewResponse> {
-      if (!memoryOverviewService) {
-        throw new Error('Memory overview service is not provided');
-      }
+      const response = (await mlClient.getMemoryStats()) as MemoryStatsResponse;
```

Comment:
Can you please raise an issue in https://github.com/elastic/elasticsearch-specification to add the correct types for this endpoint.

Reply:
https://github.com/elastic/elasticsearch-specification/issues/1538